From 2add112a195211e595ca4133f319373c68622852 Mon Sep 17 00:00:00 2001
From: Fang Wang <fang.c.wang@intel.com>
Date: Mon, 5 Jan 2015 13:56:09 +0800
Subject: [PATCH 203/210] ov5693-ecs: Correcting sequence of reading/writing
 registers

Writing registers before getting sensor integration factor.
Otherwise, driver will get incorrect atomisp_sensor_mode_data.

Tracked-On:
https://hsdes.intel.com/home/default.html#article?id=1503900502
Change-Id: Ied849d84779472a3f7626f87adfe1a6b51d3b66a
Signed-off-by: Fang Wang <fang.c.wang@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/34877
Reviewed-by: Bin B Han <bin.b.han@intel.com>
Approver: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
---
 .../external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c b/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c
index 33b79ac..c5fbf99 100755
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c
@@ -1572,6 +1572,10 @@ static int ov5693_s_mbus_fmt(struct v4l2_subdev *sd,
 		return -EINVAL;
 	}
 
+	ret = startup(sd);
+	if (ret)
+		dev_err(&client->dev, "ov5693 startup err\n");
+
 	ret = ov5693_get_intg_factor(client, ov5693_info,
 					&ov5693_res[dev->fmt_idx]);
 	if (ret) {
@@ -1579,10 +1583,6 @@ static int ov5693_s_mbus_fmt(struct v4l2_subdev *sd,
 		goto err;
 	}
 
-	ret = startup(sd);
-	if (ret)
-		dev_err(&client->dev, "ov5693 startup err\n");
-
 	ov5693_info->metadata_width = fmt->width * 10 / 8;
 	ov5693_info->metadata_height = 1;
 	ov5693_info->metadata_effective_width = &ov5693_embedded_effective_size;
-- 
1.9.1

